indigo/dataqualityreport.py

```diff
@@ -235,42 +235,97 @@ def _filter_fund_ids_that_do_not_exist_in_database(list_of_ids):
     return out
 
 
-def get_single_field_statistics_across_all_projects_for_field(field):
-    """The single part means a field that only has one value - not a list."""
-
-    field_bits = ["'" + i + "'" for i in field["key"].split("/") if i]
-    sql_where_for_field = "data_public::json->" + "->".join(field_bits)
-    sql_where_for_field_as_text = (
-        "trim(CAST(" + sql_where_for_field + " as text), '\" ')"
-    )
-
-    with connection.cursor() as cursor:
-        # How many public projects?
-        cursor.execute(
-            "select count(*) as c from indigo_project WHERE status_public=True"
-        )
-        count_public_projects = cursor.fetchone()[0]
-        # How many public projects have a value for this field?
-        cursor.execute(
-            "select count(*) as c from indigo_project WHERE status_public=True "
-            + "AND "
-            + sql_where_for_field
-            + " IS NOT NULL "
-            + "AND "
-            + sql_where_for_field_as_text
-            + " != '' "
-            + "AND "
-            + sql_where_for_field_as_text
-            + " != 'null' "
+class DataQualityReportForAllProjects:
+    def __init__(self):
+        # Cache this per request to make sure we get fresh data but don't repeat the same query tons when building the file to download
+        self.cache_count_public_projects = None
+
+    def _get_count_public_projects(self):
+        if self.cache_count_public_projects is None:
+            with connection.cursor() as cursor:
+                cursor.execute(
+                    "select count(*) as c from indigo_project WHERE status_public=True"
+                )
+                self.cache_count_public_projects = cursor.fetchone()[0]
+        return self.cache_count_public_projects
```

Comment:
This would maybe be nicer with a cache decorator https://docs.python.org/3/library/functools.html#functools.cached_property

Reply:
I will try to bear in mind for next time, thanks!